include/boost/numeric/ublas/operation/begin.hpp

```diff
@@ -28,289 +28,289 @@ namespace boost { namespace numeric { namespace ublas {
-    }
-
-
-    /**
-     * \brief A const iterator to the first element of the given vector expression.
-     * \tparam ExprT A model of VectorExpression type.
-     * \param e A vector expression.
-     * \return A const iterator to the first element of the given vector expression.
-     */
-    template <typename ExprT>
-    BOOST_UBLAS_INLINE
-    typename ExprT::const_iterator begin(vector_expression<ExprT> const& e)
-    {
-        return detail::begin_impl<typename ExprT::type_category>::apply(e());
-    }
-
-
-    /**
-     * \brief An iterator to the first element of the given matrix expression
-     *  according to its orientation.
-     * \tparam DimTagT A dimension tag type (e.g., tag::major).
-     * \tparam ExprT A model of MatrixExpression type.
-     * \param e A matrix expression.
-     * \return An iterator to the first element of the given matrix expression
-     *  according to its orientation.
-     */
-    template <typename TagT, typename ExprT>
-    BOOST_UBLAS_INLINE
-    typename iterator_type<ExprT,TagT>::type begin(matrix_expression<ExprT>& e)
-    {
-        return detail::begin_impl<typename ExprT::type_category, TagT, typename ExprT::orientation_category>::apply(e());
-    }
-
-
-    /**
-     * \brief A const iterator to the first element of the given matrix expression
-     *  according to its orientation.
-     * \tparam TagT A dimension tag type (e.g., tag::major).
-     * \tparam ExprT A model of MatrixExpression type.
-     * \param e A matrix expression.
-     * \return A const iterator to the first element of the given matrix expression
-     *  according to its orientation.
-     */
-    template <typename TagT, typename ExprT>
-    BOOST_UBLAS_INLINE
-    typename const_iterator_type<ExprT,TagT>::type begin(matrix_expression<ExprT> const& e)
-    {
-        return detail::begin_impl<typename ExprT::type_category, TagT, typename ExprT::orientation_category>::apply(e());
-    }
-
-
-    /**
-     * \brief An iterator to the first element over the dual dimension of the given
-     *  iterator.
-     * \tparam IteratorT A model of Iterator type.
-     * \param it An iterator.
-     * \return An iterator to the first element over the dual dimension of the given
-     *  iterator.
-     */
-    template <typename IteratorT>
-    BOOST_UBLAS_INLINE
-    typename IteratorT::dual_iterator_type begin(IteratorT& it)
-    {
-        return it.begin();
-    }
-
-
-    /**
-     * \brief A const iterator to the first element over the dual dimension of the
-     *  given iterator.
-     * \tparam IteratorT A model of Iterator type.
-     * \param it An iterator.
-     * \return A const iterator to the first element over the dual dimension of the
-     *  given iterator.
-     */
-    template <typename IteratorT>
-    BOOST_UBLAS_INLINE
-    typename IteratorT::dual_iterator_type begin(IteratorT const& it)
-    {
-        return it.begin();
-    }
+		    /**
```

Comment:
It looks like the only change to this file is to change the number of spaces in indentation, right ?
I would prefer not to add such changes as they are independent of your work on tensor support.
(In addition, I really think this is way too much indentation, as it doesn't leave much room for actual content, if we want to limit the line length.)

Reply:
the indentation is a tab and github's edit decides to put 4 spaces for one tab. so you can adjust it as you want on any ide and text editor.

Reply:
Ah, but we shouldn't use tabs at all ! (Really, all this adds to my argument not to mess with existing code formatting in the context of the current PR.)

Reply:
okay. no tabs any more :-)